src/Service.Tests/Unittests/RuntimeConfigLoaderJsonDeserializerTests.cs

```diff
@@ -106,6 +106,51 @@ public class RuntimeConfigLoaderJsonDeserializerTests
             }
         }
 
+        /// <summary>
+        /// Tests the parsing of the runtime configuration with environment variables based on replaceEnvVar parameter.
+        /// </summary>
+        /// <param name="replaceEnvVar">A boolean indicating whether to replace environment variables in the configuration.</param>
+        [DataTestMethod]
+        [DataRow(false, DisplayName = "Do not replace environment variables.")]
+        [DataRow(true, DisplayName = "Replace environment variables.")]
+        public void TestConfigParsingWithEnvVarReplacement(bool replaceEnvVar)
+        {
+            SetEnvironmentVariablesFromText(_environmentFileContent);
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(
+              _configWithEnvVar, out RuntimeConfig runtimeConfig, replaceEnvVar: replaceEnvVar));
+            Assert.AreEqual(runtimeConfig.DataSource.ConnectionString, GetEnvironmentVariableIfReplaced("DATABASE_CONNECTION_STRING", replaceEnvVar));
+            Assert.AreEqual(runtimeConfig.DataSource.Options["database"].ToString(), GetEnvironmentVariableIfReplaced("DATABASE_NAME", replaceEnvVar));
+            Assert.AreEqual(runtimeConfig.DataSource.Options["schema"].ToString(), GetEnvironmentVariableIfReplaced("GRAPHQL_SCHEMA_PATH", replaceEnvVar));
+            Assert.AreEqual(runtimeConfig.Runtime.Host.Authentication.Jwt.Audience, GetEnvironmentVariableIfReplaced("JWT_AUDIENCE", replaceEnvVar));
+            Assert.AreEqual(runtimeConfig.Runtime.Host.Authentication.Jwt.Issuer, GetEnvironmentVariableIfReplaced("JWT_ISSUER", replaceEnvVar));
+            Assert.AreEqual(runtimeConfig.Entities["Book"].Source.Object, GetEnvironmentVariableIfReplaced("ENTITY_SOURCE", replaceEnvVar));
+            Assert.AreEqual(runtimeConfig.Entities["Book"].Rest.Path, GetEnvironmentVariableIfReplaced("ENTITY_REST_PATH", replaceEnvVar));
+            Assert.AreEqual(runtimeConfig.Entities["Book"].Permissions[0].Role, GetEnvironmentVariableIfReplaced("ENTITY_ROLE", replaceEnvVar));
+
+            ClearEnvironmentVariablesFromText(_environmentFileContent);
+        }
+
+        /// <summary>
+        /// Retrieves the value of an environment variable if replacement is enabled, otherwise returns a placeholder string.
+        /// </summary>
+        /// <param name="envVarName">The name of the environment variable.</param>
+        /// <param name="replaceEnvVar">A boolean indicating whether to replace the environment variable with its value.</param>
+        /// <returns>
+        /// If replacement is enabled, the value of the environment variable. 
+        /// Otherwise, a placeholder string in the format "@env('variableName')".
+        /// </returns>
+        private static string GetEnvironmentVariableIfReplaced(string envVarName, bool replaceEnvVar)
```

Comment:
Name would better describe what function does by stating it's returning an expected value.
`GetExpectedPropertyValue(envVarName, replaceEnvVar)`